### ISSUE TITLE
Recompiling opencv every time on Mac arm64

### ISSUE THREAD

**Issue by jzimmerman135** [twistedfall/opencv-rust#476]
Hi I have a program that is very small, but any time I try to change anything, even just one line of a string literal that is not in an opencv call, will cause the entire opencv library to recompile. 

It happens on cargo check as well, having 40 second build time is making the library pretty unusable for me. Is there any way around this, is this a problem with opencv or just the nature of using these bindings. 

Other FFI based libraries don't really have this issue for me though.

**Comment by twistedfall** [comment]
That's definitely not supposed to happen. Can you please provide the full build log as produced by `cargo build -vv`?

**Comment by jzimmerman135** [comment]
I put the full log in an attached file, since it's very long. But it looks like it invokes clang to build opencv every time. Keep in mind this is the complete log after **deleting** a file then running `cargo b -vv`, so it really shouldn't need to be compiling anything. Thanks!

[buildlog.txt](https://github.com/twistedfall/opencv-rust/files/12100257/buildlog.txt)

**Comment by jzimmerman135** [comment]
One more thing I noticed, it even compiles opencv when I do `cargo doc --open`

**Comment by jzimmerman135** [comment]
Another update, I now notice that just by adding opencv as a dependency, regardless of whether it is called anywhere, I get a 40 second build time.

**Comment by twistedfall** [comment]
I couldn't find anything obviously wrong in the log. Could it be that one of the following environment variables change its value between the builds?
```
AR
AR_aarch64-apple-darwin
AR_aarch64_apple_darwin
ARFLAGS
ARFLAGS_aarch64-apple-darwin
ARFLAGS_aarch64_apple_darwin
CMAKE_PREFIX_PATH
CRATE_CC_NO_DEFAULTS
CXX
CXX_aarch64-apple-darwin
CXX_aarch64_apple_darwin
CXXFLAGS
CXXFLAGS_aarch64-apple-darwin
CXXFLAGS_aarch64_apple_darwin
CXXSTDLIB
CXXSTDLIB_aarch64-apple-darwin
CXXSTDLIB_aarch64_apple_darwin
DOCS_RS
HOST_AR
HOST_ARFLAGS
HOST_CXX
HOST_CXXFLAGS
HOST_CXXSTDLIB
HOST_PKG_CONFIG
HOST_PKG_CONFIG_LIBDIR
HOST_PKG_CONFIG_PATH
HOST_PKG_CONFIG_SYSROOT_DIR
OCVRS_DOCS_GENERATE_DIR
OPENCV4_DYNAMIC
OPENCV4_NO_PKG_CONFIG
OPENCV4_STATIC
OPENCV_CMAKE_BIN
OPENCV_CMAKE_NAME
OpenCV_DIR
OPENCV_DISABLE_PROBES
OPENCV_INCLUDE_PATHS
OPENCV_LINK_LIBS
OPENCV_LINK_PATHS
OPENCV_PACKAGE_NAME
OPENCV_PKGCONFIG_NAME
OPENCV_VCPKG_NAME
PKG_CONFIG
PKG_CONFIG_aarch64-apple-darwin
PKG_CONFIG_aarch64_apple_darwin
PKG_CONFIG_ALL_DYNAMIC
PKG_CONFIG_ALL_STATIC
PKG_CONFIG_LIBDIR
PKG_CONFIG_LIBDIR_aarch64-apple-darwin
PKG_CONFIG_LIBDIR_aarch64_apple_darwin
PKG_CONFIG_PATH
PKG_CONFIG_PATH_aarch64-apple-darwin
PKG_CONFIG_PATH_aarch64_apple_darwin
PKG_CONFIG_SYSROOT_DIR
PKG_CONFIG_SYSROOT_DIR_aarch64-apple-darwin
PKG_CONFIG_SYSROOT_DIR_aarch64_apple_darwin
VCPKG_ROOT
VCPKGRS_DYNAMIC
```
And additionally, did you use the previous version of the crate? And if so did you ever experience this behavior before? You can also try doing a `cargo clean` if you haven't it done before to see if it would fix the issue.

**Comment by jzimmerman135** [comment]
Hey, sorry to be MIA, didn't have time to work on this for a little while.

Finally started, messing around with things, I managed to break opencv-rust build step completely, but then started getting #462.

In the end I added the `clang-runtime` feature and it solved everything! I think this might have worked from the start too, since when I uninstall everything and start over it works too. 

Thanks for help!
